sound/soc/sof/loader.c

```diff
@@ -596,8 +589,13 @@ int snd_sof_parse_module_memcpy(struct snd_sof_dev *sdev,
 				block->size);
 			return -EINVAL;
 		}
-		snd_sof_dsp_block_write(sdev, bar, offset,
-					block + 1, block->size);
+		ret = snd_sof_dsp_block_write(sdev, block->type, offset,
+					      block + 1, block->size);
+		if (ret < 0) {
+			dev_err(sdev->dev, "error: no BAR mapping for block type 0x%x\n",
```

Comment:
shouldn't the error message be updated now too?

Reply:
I have already changed it to "error: write to block type 0x%x failed\n"